tests/dispatchers/sync.test.ts

```diff
@@ -0,0 +1,179 @@
+import * as fs from 'fs/promises';
+import { handler } from '../../src/commands/dispatchers/sync';
+import * as dispatcherRepository from '../../src/repositories/dispatchers';
+import { generateMailAction, generateTaskAction } from '../__helpers__/actions';
+import { generateDispatcher, generateMinimalDispatcher } from '../__helpers__/dispatchers';
+
+describe('Dispatchers - Sync', () => {
+  let updateDispatcherSpy;
+  let updateActionSpy;
+  let deleteActionSpy;
+
+  const existingDispatcher = generateDispatcher();
+
+  beforeAll(() => {
+    // TODO: Move these somewhere else?
+    jest.spyOn(dispatcherRepository, 'findAll')
+      .mockImplementation(() => Promise.resolve([existingDispatcher, generateDispatcher()]));
+
+    updateDispatcherSpy = jest.spyOn(dispatcherRepository, 'update')
+      .mockImplementation(() => Promise.resolve({ affectedRecords: 1 }));
+
+    updateActionSpy = jest.spyOn(dispatcherRepository, 'updateAction')
+      .mockImplementation(() => Promise.resolve({ affectedRecords: 1 }));
+
+    deleteActionSpy = jest.spyOn(dispatcherRepository, 'deleteAction')
+      .mockImplementationOnce(() => Promise.resolve({ affectedRecords: 1 }));
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('Throws for a Dispatcher without a name', async () => {
+    const dispatcher = generateDispatcher({
+      name: undefined,
+    });
+
+    jest.spyOn(fs, 'readFile')
+      .mockImplementationOnce(() => Promise.resolve(JSON.stringify([dispatcher])));
+
+    const error = await handler({ sdk: undefined, file: '' })
```

Comment:
(minor) I would write the test like this:
    `await expect(handler({ sdk: undefined, file: '' })).rejects.toThrow('Dispatcher name is a required field');`
    
It gives more information in the error reporting then.
e.g: 
<img width="957" alt="Screenshot 2023-09-19 at 14 28 38" src="https://github.com/ExtraHorizon/exh-cli/assets/93595259/9a5ece75-0eb0-467c-92fc-9cda61a27ff4">
vs
<img width="811" alt="Screenshot 2023-09-19 at 14 29 19" src="https://github.com/ExtraHorizon/exh-cli/assets/93595259/1e4efe76-32a1-4137-97cc-a25654331998">
 
 Same for some of the following tests.